tools/gcs-bucket-mover/gcs_bucket_mover/bucket_mover_tester.py

```diff
@@ -29,184 +26,190 @@
-
-    # Bucket Notification
-    notification = storage.notification.BucketNotification(
-        bucket,
-        parsed_args.test_topic_name,
-        custom_attributes={'myKey': 'myValue'},
-        event_types=['OBJECT_FINALIZE', 'OBJECT_DELETE'],
-        payload_format='JSON_API_V1')
-    notification.create()
-
-    return bucket
+  """Creates the test bucket.
+
+  Also sets up lots of different bucket settings to make sure they can be moved.
+
+  Args:
+      storage_client: The storage client object used to access GCS
+      bucket_name: The name of the bucket to create
+      parsed_args: the configargparser parsing of command line options
+
+  Returns:
+      The bucket object that has been created in GCS
+  """
+
+  bucket = storage.Bucket(client=storage_client, name=bucket_name)
+  # Requester pays
+  bucket.requester_pays = False
+  # CORS
+  policies = bucket.cors
+  policies.append({"origin": ["/foo"]})
+  policies[0]["maxAgeSeconds"] = 3600
+  bucket.cors = policies
+  # KMS Key - When a custom KMS key is set up, uncomment the line below to test it
+  # bucket.default_kms_key_name = parsed_args.test_default_kms_key_name
+  # Labels
+  bucket.labels = {"colour": "red", "flavour": "cherry"}
+  # Object Lifecycle Rules
+  bucket.lifecycle_rules = [{
+    "action": {
+      "type": "Delete"
+    },
+    "condition": {
+      "age": 365
+    }
+  }]
+  # Location
+  bucket.location = parsed_args.test_bucket_location
+  # Storage Class
+  bucket.storage_class = parsed_args.test_storage_class
+  # File Versioning
+  # Setting this to True means we can't delete a non-empty bucket with the CLI in one
+  # bucket.delete command
+  bucket.versioning_enabled = False
+  # Access Logs
+  bucket.enable_logging(parsed_args.test_logging_bucket,
+                        parsed_args.test_logging_prefix)
+
+  bucket.create()
+
+  # IAM Policies
+  policy = bucket.get_iam_policy()
+  # Uncomment the line below to view the existing IAM policies
+  # print(json.dumps(policy.to_api_repr(), indent=4, sort_keys=True))
+  policy["roles/storage.admin"].add("user:" + parsed_args.test_email_for_iam)
+  bucket.set_iam_policy(policy)
+  # ACLs
+  bucket.acl.user(parsed_args.test_email_for_iam).grant_read()
+  bucket.acl.save()
+  # Default Object ACL
+  bucket.default_object_acl.user(parsed_args.test_email_for_iam).grant_read()
+  bucket.default_object_acl.save()
+
+  bucket.update()
+
+  # Bucket Notification
+  notification = storage.notification.BucketNotification(
+    bucket,
+    parsed_args.test_topic_name,
+    custom_attributes={"myKey": "myValue"},
+    event_types=["OBJECT_FINALIZE", "OBJECT_DELETE"],
+    payload_format="JSON_API_V1",
+  )
+  notification.create()
+
+  return bucket
 
 
 def _get_random_bucket_name():
-    """Generate a random bucket name for testing purposes"""
-    return 'bucket_mover_' + str(uuid.uuid4())
+  """Generate a random bucket name for testing purposes"""
+  return "bucket_mover_" + str(uuid.uuid4())
```

Comment:
please standardize usage of single or double quotes. I see both being used in the PR

Reply:
@markueez Please address the comments and the conflict in requirements.txt, and I'll try to get this merged.